nutils/function.py

```diff
@@ -2397,6 +2398,7 @@ class DerivativeHelper( ArrayFunc ):
   def __init__( self, shape, axes ):
     self._axes = tuple(axes)
     assert all(0 <= axis < len(shape) for axis in self._axes)
```

Comment:
Since `shape` has been replaced by `axes` in `ArrayFunc._derivative` I'm not sure if it's necessary to store the same `axes` as an attribute here.